runtime-sdk/src/modules/consensus_accounts/mod.rs

```diff
@@ -0,0 +1,422 @@
+//! Consensus accounts module.
+use std::str::FromStr;
+
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+use oasis_core_runtime::{common::cbor, consensus::staking::Account as ConsensusAccount};
+
+use crate::{
+    context::{Context, DispatchContext, TxContext},
+    error::{self, Error as _},
+    module,
+    module::{CallableMethodInfo, Module as _, QueryMethodInfo},
+    modules,
+    types::{
+        address::Address,
+        message::{MessageEvent, MessageEventHookInvocation},
+        token,
+        transaction::CallResult,
+    },
+};
+
+#[cfg(test)]
+mod test;
+pub mod types;
+
+/// Unique module name.
+const MODULE_NAME: &str = "consensus_accounts";
+
+#[derive(Error, Debug, oasis_runtime_sdk_macros::Error)]
+pub enum Error {
+    #[error("invalid argument")]
+    #[sdk_error(code = 1)]
+    InvalidArgument,
+    #[error("invalid denomination")]
+    #[sdk_error(code = 2)]
+    InvalidDenomination,
+    #[error("withdraw: insufficient runtime balance")]
+    #[sdk_error(code = 3)]
+    InsufficientWithdrawBalance,
+    #[error("core error")]
+    #[sdk_error(code = 4)]
+    CoreError(#[from] modules::core::Error),
+}
+
+/// Events emitted by the consensus module (none so far).
+#[derive(Debug, Serialize, Deserialize, oasis_runtime_sdk_macros::Event)]
+#[serde(untagged)]
+pub enum Event {}
+
+/// Parameters for the consensus module.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
+pub struct Parameters {
+    pub consensus_denomination: token::Denomination,
+}
+
+impl Default for Parameters {
+    fn default() -> Self {
+        Self {
+            // TODO: get this from consensus? (token symbol?).
```

Comment:
I think it makes sense to ensure that the consensus token is named consistently in all ParaTimes. On the other hand we also don't want to constantly query the consensus layer for this denomination as it is a pretty constant thing (it cannot actually be changed in the consensus layer unless there is a hard fork).

It could be queried during initialization (in `MigrationHandler`) and set as a parameter here?